doi-transformer/src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java

```diff
@@ -117,35 +112,55 @@ public Publication toPublication(CrossRefDocument document,
-                            .withMainTitle(extractTitle(document))
-                            .withAlternativeTitles(extractAlternativeTitles(document))
-                            .withAbstract(extractAbstract(document))
-                            .withLanguage(extractLanguage(document))
-                            .withNpiSubjectHeading(extractNpiSubjectHeading())
-                            .withTags(extractSubject(document))
-                            .withReference(extractReference(document))
-                            .withMetadataSource(extractMetadataSource(document))
-                            .build())
-                    .build();
+                .withCreatedDate(extractInstantFromCrossrefDate(document.getCreated()))
+                .withModifiedDate(extractInstantFromCrossrefDate(document.getDeposited()))
+                .withPublishedDate(extractInstantFromCrossrefDate(document.getIssued()))
+                .withOwner(owner)
+                .withDoi(extractDOI(document)) // Cheating by using URL not DOI ?
+                .withIdentifier(new SortableIdentifier(identifier.toString()))
+                .withPublisher(extractAndCreatePublisher(document))
+                .withStatus(DEFAULT_NEW_PUBLICATION_STATUS)
+                .withIndexedDate(extractInstantFromCrossrefDate(document.getIndexed()))
+                .withLink(extractFulltextLinkAsUri(document))
+                .withProjects(createProjects())
+                .withFileSet(createFilseSet())
+                .withEntityDescription(new EntityDescription.Builder()
+                                           .withContributors(toContributors(document))
+                                           .withDate(extractIssuedDate(document))
+                                           .withMainTitle(extractTitle(document))
+                                           .withAlternativeTitles(extractAlternativeTitles(document))
+                                           .withAbstract(extractAbstract(document))
+                                           .withLanguage(extractLanguage(document))
+                                           .withNpiSubjectHeading(extractNpiSubjectHeading())
+                                           .withTags(extractSubject(document))
+                                           .withReference(extractReference(document))
+                                           .withMetadataSource(extractMetadataSource(document))
+                                           .build())
+                .build();
         }
         throw new IllegalArgumentException(INVALID_ENTRY_ERROR);
     }
 
+    protected List<Contributor> toContributors(CrossRefDocument document) {
+
+        List<Contributor> contributors = new ArrayList<>();
+        contributors.addAll(toContributorsWithRole(document.getAuthor(), Role.CREATOR));
+        contributors.addAll(toContributorsWithRole(document.getEditor(), Role.EDITOR));
```

Comment:
We also have a role called Supervisor should that have been included here somehow?